rig/place_and_route/place/sa/__init__.py

```diff
@@ -0,0 +1,19 @@
+"""An experimental simulated-annealing based placer.
+
+The annealing algorithm is broken into two components: the high-level algorithm
+implementation :py:func:`~rig.place_and_route.place.sa.place` and a simulated
+annealing placement :py:class:`~rig.place_and_route.place.sa.kernel.Kernel`.
+
+The algorithm takes care of initial placement, handles special-case "trivial"
+placement problems where no placer effort is requried, handles details of
+:py:mod:`~rig.place_and_route.constraints`, and manages the annealing scheudle.
+
+The kernel is responsible for performing the kernel of the annealing operation:
+swapping vertices, evaluating the change in cost and reverting (some) bad
+swaps. Since this is the most performance-sensitive part of the algorithm, its
+implementation may be swapped for more efficient implementations as required. A
+portable, but slow, kernel written in Python is included in
+:py:class:`~rig.place_and_route.place.sa.python_kernel.PythonKernel`.
```

Comment:
Unfortunately not since I've not managed to get pypy and numpy to play nice together... Back when I was playing with the algorithm, however, PyPy got you 4-8x the performance of native Python. I don't expect that this will have changed too much.